### PR TITLE
Have `die_with_error` use `fmt::Display` rather than Debug

### LINKED ISSUE #1099: die_with_error should use fmt::Display instead of fmt::Debug
https://github.com/ethcore/parity/blob/dcc695dda5b77fa0199a6061802d086ba59ab329/parity/die.rs#L32

Additionally it should trace error `trace!("{:?}", e)`

### PR DESCRIPTION
**Pull request by rphmeier**
cc @debris 

Fixes #1099 

Please look at the error messages I wrote for accuracy and refinement!

### CODE REVIEW

File: util/src/error.rs [hunk 1]
```diff
@@ -78,6 +113,27 @@ pub struct OutOfBounds<T: fmt::Debug> {
 	pub found: T,
 }
 
+impl<T: fmt::Debug + fmt::Display> fmt::Display for OutOfBounds<T> {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		let msg = match (self.min.as_ref(), self.max.as_ref()) {
+			(Some(min), Some(max)) => {
```
Comment (gavofyork) [inline util/src/error.rs]: all single-statement arms - no need for `{`s

File: ethcore/src/error.rs [hunk 2]
```diff
@@ -132,6 +258,24 @@ pub enum ImportError {
 	KnownBad,
 }
 
+impl fmt::Display for ImportError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		let msg = match self {
+			&ImportError::AlreadyInChain => {
```
Comment (gavofyork) [inline ethcore/src/error.rs]: single-statement arms - `{`s superfluous

File: ethcore/src/error.rs [hunk 3]
```diff
@@ -121,6 +158,95 @@ pub enum BlockError {
 	UnknownUncleParent(H256),
 }
 
+impl fmt::Display for BlockError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		use self::BlockError::*;
+
+		let msg = match *self {
+			TooManyUncles(ref oob) => {
```
Comment (gavofyork) [inline ethcore/src/error.rs]: rather a lot of unneeded `{`s...

File: ethcore/src/error.rs [hunk 4]
```diff
@@ -132,6 +258,24 @@ pub enum ImportError {
 	KnownBad,
 }
 
+impl fmt::Display for ImportError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		let msg = match self {
```
Comment (tomusdrw) [inline ethcore/src/error.rs]: `match *self` -> https://github.com/Manishearth/rust-clippy/wiki#match_ref_pats

File: ethcore/src/error.rs [hunk 5]
```diff
@@ -60,6 +60,43 @@ pub enum TransactionError {
 	InvalidGasLimit(OutOfBounds<U256>),
 }
 
+impl fmt::Display for TransactionError {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		use self::TransactionError::*;
+		let msg = match *self {
+			AlreadyImported => {
+				"Already imported".into()
```
Comment (tomusdrw) [inline ethcore/src/error.rs]: :+1: for `into` :)

**Comment by rphmeier** [comment]
I am working on addressing the style concerns at the moment.

File: util/src/error.rs [hunk 6]
```diff
@@ -57,6 +67,24 @@ pub enum UtilError {
...
+			UtilError::Crypto(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::StdIo(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::Io(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::AddressParse(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::AddressResolve(Some(ref err)) => f.write_fmt(format_args!("{}", err)),
+			UtilError::AddressResolve(_) => f.write_str("Failed to resolve network address."),
+			UtilError::FromHex(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::BaseData(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::Network(ref err) => f.write_fmt(format_args!("{}", err)),
+			UtilError::Decoder(ref err) => f.write_fmt(format_args!("{}", err)),
```
Comment (rphmeier) [inline util/src/error.rs]: DecoderError has a Display impl which just forwards to its derived Debug impl. This can be rewritten as well.

**Pull request merged**